src/platform/unix/mod.rs

```diff
@@ -1162,18 +1162,36 @@ impl UnixCmsg {
                 }
             },
             BlockingMode::Timeout(duration) => {
+                #[cfg(target_os = "linux")]
                 let events = libc::POLLIN | libc::POLLPRI | libc::POLLRDHUP;
+
+                // It's rather unfortunate that Rust's libc doesn't know that
+                // FreeBSD has POLLRDHUP, but in the mean time we can add
+                // the value ourselves.
+                // https://cgit.freebsd.org/src/tree/sys/sys/poll.h#n72
+                #[cfg(target_os = "freebsd")]
+                let events = libc::POLLIN | libc::POLLPRI | 0x4000;
+
                 let mut fd = [libc::pollfd {
                     fd,
                     events,
                     revents: 0,
                 }];
+                #[cfg(target_os = "linux")]
                 let result = libc::poll(
                     fd.as_mut_ptr(),
                     fd.len() as libc::c_ulong,
                     duration.as_millis().try_into().unwrap_or(-1),
                 );
 
+		// FreeBSD's nfds_t is an unsigned integer.
+		#[cfg(target_os = "freebsd")]
+                let result = libc::poll(
+                    fd.as_mut_ptr(),
+                    fd.len() as libc::c_uint,
```

Comment:
We should be able to use `as _` and avoid any platform-specific code.

Reply:
Fixed up.